fix: update integration tests to use 2-element variable selectors

farion1231 opened the pull request:
## Problem

After the VariablePool refactoring in commit `577062b93`, 24 integration tests are failing with the error:

```
ValueError: Invalid selector: expected 2 elements (node_id, variable_name), got 3 elements
```

This is blocking multiple PRs from being merged as CI consistently fails on the integration test phase.

## Root Cause

The VariablePool API was refactored to only accept 2-element variable selectors, but the integration tests were not updated to match this change.

**What changed:**
- **Before**: `["node_id", "node_title", "variable_name"]` ✅ (3 elements)
- **After**: `["node_id", "variable_name"]` ✅ (2 elements only)

**Why the original refactor PR passed CI:**
1. Unit tests were properly updated by the author
2. Integration tests still used the old 3-element format
3. Integration tests likely failed silently due to Docker environment issues or CI configuration problems

## Solution

Remove the middle element (node_title) from all variable selectors in integration tests, keeping only the essential node_id and variable_name.

**Example fixes:**
```python
# Before (❌)
variable_pool.add(["code", "123", "args1"], 1)
"value_selector": ["1", "123", "args1"]

# After (✅)
variable_pool.add(["code", "args1"], 1)  
"value_selector": ["1", "args1"]
```

## Files Changed

- `api/tests/integration_tests/workflow/nodes/test_code.py`
- `api/tests/integration_tests/workflow/nodes/test_http.py`
- `api/tests/integration_tests/workflow/nodes/test_template_transform.py`
- `api/tests/integration_tests/workflow/nodes/test_parameter_extractor.py`
- `api/tests/integration_tests/workflow/nodes/test_tool.py`

## Impact

- ✅ Fixes 24 failing integration tests
- ✅ Unblocks CI for all pending PRs
- ✅ Maintains test semantic correctness (node_id is sufficient for identification)
- ✅ Aligns tests with the new VariablePool architecture

This fix restores the integration test suite to working condition and allows normal development workflow to resume. 

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

farion1231 committed:
fix: update integration tests to use 2-element variable selectors

- Remove obsolete node title from variable selectors
- Convert [node_id, title, var_name] to [node_id, var_name] format
- Align with new VariablePool implementation that expects 2-element selectors
- Fix 24 failing integration tests across 5 test files

Files updated:
- test_code.py: Fix value_selector and variable_pool.add calls
- test_http.py: Fix variable_pool.add calls
- test_template_transform.py: Fix value_selector and variable_pool.add calls
- test_parameter_extractor.py: Fix variable_pool.add calls
- test_tool.py: Fix variable_pool.add calls

farion1231 committed:
fix: update HTTP test template references to match 2-element selectors

- Replace {{#a.b123.args1#}} with {{#a.args1#}} in HTTP node tests
- Replace {{#a.b123.args2#}} with {{#a.args2#}} in HTTP node tests
- Ensure template variable resolution aligns with new VariablePool format
- Test fix for 3 failing HTTP integration tests

QuantumGhost reviewed (dismissed):
LGTM

QuantumGhost commented on api/tests/integration_tests/workflow/nodes/test_http.py:
```diff
@@ -171,7 +171,7 @@ def test_template(setup_http_mock):
                 "desc": "",
                 "method": "get",
-                "url": "http://example.com/{{#a.b123.args2#}}",
+                "url": "http://example.com/{{#a.args2#}}",
```
I have a suggestion to enhance the test coverage:

At `api/tests/integration_tests/workflow/nodes/test_http.py:174`, I recommend adding an additional test case to verify variable selector functionality with nested object properties.

**Suggested enhancement:**

Add a test case with a nested object structure by:

1. Adding a nested object variable:**
   ```python
   variable_pool.add([("a", "args3")], {"nested": "nested_value"})
   ```

2. **Updating the URL configuration to test nested property access:**
   ```python
   "url": "http://example.com/{{#a.args2#}}/{{#a.args3.nested#}}"
   ```

This addition would ensure that the logic for extracting values from `ObjectSegment` types is covered.

farion1231 replied:
Thank you for the suggestion on enhancing test coverage with nested object properties.

  I've implemented your recommendation by adding:

  Nested object variable:
  variable_pool.add(["a", "args3"], {"nested": "nested_value"})

  Comprehensive nested property testing:
  - URL: {{#a.args2#}}/{{#a.args3.nested#}}
  - Headers: X-Header:{{#a.args3.nested#}}
  - Parameters: nested_param:{{#a.args3.nested#}}

  Test isolation:
  To prevent cross-test pollution, I created a completely independent test implementation rather than modifying the shared init_http_node() function. This ensures the new
   test doesn't affect existing test behavior.

farion1231 committed:
enhance: add nested object property test coverage for HTTP node

- Add nested object variable to test ObjectSegment functionality
- Create test_nested_object_variable_selector() to verify nested property access
- Test nested variable resolution in URL, headers, and parameters
- Verify {{#a.args3.nested#}} template syntax correctly resolves to nested values

Addresses review feedback for enhanced test coverage of variable selector functionality.

farion1231 committed:
fix: isolate nested object test to prevent cross-test pollution

- Remove args3 variable from shared init_http_node() function
- Create completely independent test_nested_object_variable_selector()
- Implement isolated variable pool setup for nested object testing
- Prevent unintended side effects on other HTTP integration tests
- Maintain reviewer-requested nested property test coverage

Resolves potential test pollution that may have caused unrelated test failures.
